src/types/index.d.ts

```diff
@@ -1,3 +1,7 @@
 interface InnerFocusable {
   focusInner: () => Promise<void>
 }
+
+interface ClonesAttributes {
+  clonedAttributes: { name: string; value: string }[] // should use State decorator
```

Comment:
```suggestion
  clonedAttributes: Record<string, any> // should use State decorator
```

This would be the correct typing, but not sure if this works well with the implementation.

Reply:
Correct. The array was history, I wonder why TS didn't complain... Anyway, your suggestion should work, but I'm not sure about the `any`. I would rather allow only string and number as values for now. I've just updated this to
```ts
  clonedAttributes: Record<string, string | number>
```

Reply:
At least boolean values should be added, as well. But as some props are for example defined as set of explicit string values (like the `mode` props) and we cannot properly define explicit types for all props here. I think `any` would be more correct.

I think TS might not have complained, because you would need to declare the `private clonedAttributes` property with an explicit type in each component. The way it's declared now, probably results in its type being `any`...?

Reply:
Using `string | number | boolean`. Updated.